### PR TITLE
tree-wide: Introduce netif feature and use it

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This gets rid of a long list of boards with network interfaces and instead let's boards (or MCUs with peripheral network interfaces) provide the netif feature.

The apps that before used the long list are not depending on the feature instead (in case of the default example, this is an optional dependency).

### Testing procedure

Ideally, no change in binaries. There may be some boards that actually do have a network interface but were not listed in `Makefile.boards.netif` that now qualify as having a network interface, if the feature was provided on MCU level (only for peripheral netifs).

### Issues/PRs references

Partially fixes https://github.com/RIOT-OS/RIOT/issues/20680

### CODE REVIEW

File: boards/common/nrf52/Makefile.features [hunk 1]
```diff
@@ -3,4 +3,6 @@ CPU = nrf52
 # Put defined MCU peripherals here (in alphabetical order)
 FEATURES_PROVIDED += periph_rtt
 FEATURES_PROVIDED += periph_timer
+
+# Put other features for this board (in alphabetical order)
```
Comment (mguetschow) [inline boards/common/nrf52/Makefile.features:7]:
```suggestion

# Put other features for this board (in alphabetical order)
FEATURES_PROVIDED += netif
```

**Review by mguetschow — CHANGES_REQUESTED** [review]
Thanks for that! I haven't compared the removed list of boards vs. added `FEATURES_PROVIDED` in detail, but I'd trust you to have done that thoroughly enough - in the end it is only relevant for apps/tests using the default board features.

Just a minor addition below.

**Comment by bmewen** [comment]
Sounds good to me, I tested it with a dwm1001, an arduino nano 33 ble (and the sense one) and didn't notice any unexpected behavior on the default example.
The board list also seems to be compliant with what was existing. 

My only question is about the following line : https://github.com/RIOT-OS/RIOT/blob/0641ab2b709471719ba4e3ff5e76291b3bbb5782/examples/default/Makefile#L49, since you added a feature it should be the same as this one I think : https://github.com/RIOT-OS/RIOT/blob/0641ab2b709471719ba4e3ff5e76291b3bbb5782/examples/default/Makefile.board.dep#L1

**Review by mguetschow — APPROVED** [review]

**Comment by maribu** [comment]
Looks like the `esp32-ethernet-kit*` failed to build with `esp32_eth` and `saul_default` in use, but this was never triggered in the CI in `master` because they were not in `BOARD_PROVIDES_NETIF`.

The second commit fixes that issue.

**Review by mguetschow — APPROVED** [review]
Good catch, thanks @maribu!

**Pull request merged**